packages/cli/src/lib/setup.ts

```diff
@@ -180,35 +175,31 @@ write_status() {
         }
         | if ($error != "") then . + {error: $error} else . end
         | if ($status == "completed" or $status == "failed") then . + {completedAt: $completedAt} else . end' \\
-        > /output/status.json
+      | sudo tee /output/status.json
+
+    sudo chown $(id -u):$(id -g) /output/status.json
 }`;
   }
 
   /**
-   * Generate credential shredding and permission recovery function
+   * Generate cleanup functions
    */
   private generateCleanupFunctions(): string {
-    let isDockerRootless = true;
-    let recoverPermissions = `
-chown -R $uid:$gid /workspace || true
-chown -R $uid:$gid /output || true
-`;
+    let isDockerRootless = false;
+
     const dockerInfo = launchSync('docker', ['info', '-f', 'json']).stdout;
     if (dockerInfo) {
       const info = JSON.parse(dockerInfo.toString());
       isDockerRootless = (info?.SecurityOptions || []).some((value: string) =>
         value.includes('rootless')
       );
-      if (isDockerRootless) {
-        recoverPermissions = `
-chown -R root:root /workspace || true
-chown -R root:root /output || true
-`;
-      }
-    } else {
+    }
+
+    let recoverPermissions = '';
+    if (isDockerRootless) {
       recoverPermissions = `
-${recoverPermissions}
-echo "❌ It was not possible to identify Docker installation information on the host, project permissions might be off"
+sudo chown -R root:root /workspace || true
+sudo chown -R root:root /output || true
```

Comment:
Based on this, I assume we are no longer removing the `sudo` permissions for the `agent` user, right? This might get executed right before stopping the process.

Reply:
In this PR we are not removing ourselves from sudoers yet, although we should, I consider that a very important security change that we should do, but wanted to have everything working first.

We can remove ourselves from sudoers for all commands, but keep in the sudoers the ability to call chown only for example, or a custom binary that does just this.

Reply:
Sounds good!

Reply:
Done :)